public/js/main.js

```diff
@@ -200,3 +200,83 @@ async function updateAddress() {
   }
   
 }
+
+// Upload Photo from barista profile to Cloudinary 
+
+async function uploadProfilePhoto(){
+  try{
+    const signResponse = await fetch('/barista/signuploadform', {
+      method: "get"
+    });
+    const signData = await signResponse.json();
+    const cloudinaryURL = "https://api.cloudinary.com/v1_1/" + signData.cloudname + "/auto/upload";
+
+    const files = document.getElementById("uploadPhoto").files;
+    const formData = new FormData();
+
+    for (let i = 0; i < files.length; i++) {
+      let file = files[i];
+      formData.append("file", file);
+      formData.append("api_key", signData.apikey);
+      formData.append("timestamp", signData.timestamp);
+      formData.append("signature", signData.signature);
+      formData.append("resource_type", "image");
+      formData.append("allowed_formats", "jpg,jpeg,png,bmp,gif");
+      formData.append("eager", "c_thumb,h_150,w_150,g_face");
+      formData.append("folder", "profile_photos");
+    }
+    
+    const photoDataResponse = await fetch(cloudinaryURL, {
+      method: "POST",
+      body: formData
+    })
+    const photoData = await photoDataResponse.json()
+
+    const uploadResult = document.getElementById('photoUploadResult').appendChild(document.createElement('img'))
+    uploadResult.setAttribute('src', photoData.eager[0].secure_url)

```

Comment:
Kind of a nitpick, but I'm a little more used to seeing this:

```js
const imgNode = document.createElement('img');
imgNode.setAttribute('src', photoData.eager[0].secure_url);
document.getElementById('photoUploadResult').appendChild(imgNode);
```

In your version, the `img` tag is added to the DOM before the `src` is set. Since `img` elements require the `src` attribute, this makes your DOM invalid, even if only for a split second.